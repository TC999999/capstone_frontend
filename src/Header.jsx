const Header = () => {
  return (
    <h1 className="app-header">
      <span>W</span>
      <span>o</span>
      <span>r</span>
      <span>l</span>
      <span>d</span>
      <span>w</span>
      <span>i</span>
      <span>d</span>
      <span>e</span>
      <span> </span>
      <span>G</span>
      <span>a</span>
      <span>r</span>
      <span>a</span>
      <span>g</span>
      <span>e</span>
      <span> </span>
      <span>S</span>
      <span>a</span>
      <span>l</span>
      <span>e</span>
      <span>!</span>
    </h1>
  );
};

export default Header;
