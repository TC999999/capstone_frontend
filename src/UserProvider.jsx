import UserContext from "./UserContext.js";
import { useState, useEffect } from "react";
import marketAPI from "../api.js";

const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  //sets context for user for entire app
  const currentUser = (user) => {
    setUser(user);
  };

  // //On initial render, get user token from local storage, gets the current user from the api, and set as context
  useEffect(() => {
    async function getToken() {
      let token = localStorage.getItem("market-token");
      if (token) {
        let userInfo = await marketAPI.getCurrentUser();
        currentUser(userInfo);
      }
    }
    getToken();
  }, []);

  //These functions and user state will be used in context
  //Set the token in local storage, retrives the current user from the api, and sets the user as context on login
  const logIn = async (username, token) => {
    localStorage.setItem("market-token", token);
    let userInfo = await marketAPI.getUserInfo(username);
    currentUser(userInfo);
  };

  //remove token and user from local storage and set user context to null on logout
  const logOut = () => {
    localStorage.removeItem("market-token");
    currentUser(null);
  };

  //update current user in local storage and context
  const updateUser = async (username) => {
    let userInfo = await marketAPI.getUserInfo(username);
    currentUser(userInfo);
  };

  return (
    <UserContext.Provider
      value={{ user, currentUser, logIn, logOut, updateUser }}
    >
      {children}
    </UserContext.Provider>
  );
};

export default UserProvider;
