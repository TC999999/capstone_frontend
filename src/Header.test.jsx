import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserProvider from "./UserProvider.jsx";
import Header from "./Header.jsx";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Header />
      </UserProvider>
    </MemoryRouter>
  );
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Header />
      </UserProvider>
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});
