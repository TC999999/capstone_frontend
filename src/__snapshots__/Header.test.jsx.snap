// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <h1
    class="app-header"
  >
    <span>
      W
    </span>
    <span>
      o
    </span>
    <span>
      r
    </span>
    <span>
      l
    </span>
    <span>
      d
    </span>
    <span>
      w
    </span>
    <span>
      i
    </span>
    <span>
      d
    </span>
    <span>
      e
    </span>
    <span>
       
    </span>
    <span>
      G
    </span>
    <span>
      a
    </span>
    <span>
      r
    </span>
    <span>
      a
    </span>
    <span>
      g
    </span>
    <span>
      e
    </span>
    <span>
       
    </span>
    <span>
      S
    </span>
    <span>
      a
    </span>
    <span>
      l
    </span>
    <span>
      e
    </span>
    <span>
      !
    </span>
  </h1>
</DocumentFragment>
`;
